pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -56,16 +61,39 @@ type ProxierHealthServer struct {
 }
 
 // NewProxierHealthServer returns a proxier health http server.
-func NewProxierHealthServer(addr string, healthTimeout time.Duration) *ProxierHealthServer {
-	return newProxierHealthServer(stdNetListener{}, stdHTTPServerFactory{}, clock.RealClock{}, addr, healthTimeout)
+func NewProxierHealthServer(cidrStrings []string, port int32, healthTimeout time.Duration) *ProxierHealthServer {
+	return newProxierHealthServer(stdNetListener{}, stdHTTPServerFactory{}, clock.RealClock{}, proxyutil.RealNetwork{}, cidrStrings, port, healthTimeout)
 }
 
-func newProxierHealthServer(listener listener, httpServerFactory httpServerFactory, c clock.Clock, addr string, healthTimeout time.Duration) *ProxierHealthServer {
+func newProxierHealthServer(listener listener, httpServerFactory httpServerFactory, c clock.Clock, nw proxyutil.NetworkInterfacer, cidrStrings []string, port int32, healthTimeout time.Duration) *ProxierHealthServer {
+	var nodeIPs []net.IP
+	var err error
+	// Some cloud-providers may assign IPs to a node after kube-proxy
+	// startup. The only way to listen on those IPs is to bind the server
+	// on 0.0.0.0. To handle this case we skip filtering NodeIPs by CIDRs
+	// and listen on 0.0.0.0 if any of the given CIDRs is a zero-cidr.
+	// (ref: https://github.com/kubernetes/kubernetes/pull/126889)
+	if sets.New(cidrStrings...).HasAny(proxyutil.IPv4ZeroCIDR, proxyutil.IPv6ZeroCIDR) {
+		nodeIPs = []net.IP{net.IPv4zero}
+	} else {
+		nodeIPs, err = proxyutil.FilterInterfaceAddrsByCIDRStrings(nw, cidrStrings)
```

Comment:
`NodePortAddresses` already has special handling to recognize `0.0.0.0/0` / `::/0` and let the caller know about it (`.MatchAll()`) (because we treat this as a special case for NodePort rules too). Given that, it seems like rather than pulling out `proxyutil.FilterInterfaceAddrsByCIDRStrings`, we should just use `NodePortAddresses` here and in `server.go` too.

But maybe rename it to not be nodeport-specific? (`NodeAddressHandler`?)